modules/elprep/filter/main.nf

```diff
@@ -0,0 +1,99 @@
+process ELPREP_FILTER {
+    tag "$meta.id"
+    label 'process_high'
+
+    conda (params.enable_conda ? "bioconda::elprep=5.1.2" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/elprep:5.1.2--he881be0_0':
+        'quay.io/biocontainers/elprep:5.1.2--he881be0_0' }"
+
+    input:
+    tuple val(meta), path(bam)
+    val(run_haplotypecaller)
+    val(run_bqsr)
+    path(reference_sequences)
+    path(filter_regions_bed)
+    path(reference_elfasta)
+    path(known_sites_elsites)
+    path(target_regions_bed)
+    path(intermediate_bqsr_tables)
+    val(bqsr_tables_only)
+    val(get_activity_profile)
+    val(get_activity_regions)
+
+
+    output:
+    tuple val(meta), path("**.{bam,sam}")           ,emit: bam
+    tuple val(meta), path("*.metrics.txt")          ,optional: true, emit: metrics
+    tuple val(meta), path("*.recall")               ,optional: true, emit: recall
+    tuple val(meta), path("*.vcf.gz")               ,optional: true, emit: gvcf
+    tuple val(meta), path("*.table")                ,optional: true, emit: table
+    tuple val(meta), path("*.activity_profile.igv") ,optional: true, emit: activity_profile
+    tuple val(meta), path("*.assembly_regions.igv") ,optional: true, emit: activity_regions
+    path "versions.yml"                             ,emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def suffix = args.contains("--output-type sam") ? "sam" : "bam"
+
+    // filter args
+    if (reference_sequences) {
+        args += " --replace-reference-sequences ${reference_sequences}"
+    }
```

Comment:
Any reason why all of these are not in 

```suggestion
reference_sequences ? args += " --replace-reference-sequences ${reference_sequences}" : ""
```

notation? Would make this easier to read ;)

Reply:
probably because I'm not entirely comfortable with the syntax yet. I'll modify it!